plugins/pack-plugin.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+# this is a bash helper for packing a plugin from its source files.
+# Usage: pack-plugin.sh plugin_dir
+# where plugin_dir is the directory where the plugin exists
+
+set -o errexit
+set -o errtrace
+set -o nounset
+set -o pipefail
+#set -o xtrace
+
+PLUGIN_DIR_INPUT=${1:-}
+ZIP="$(which zip)"
+
+if [[ ! -x "$ZIP" ]]; then
+    echo "Cannot find executable: zip"
+    exit 99
+fi
+
+if [[ "$PLUGIN_DIR_INPUT" = "" ]]; then
+    echo "Usage: $0 plugin_dir"
+    exit 1
+fi
+PLUGIN_DIR="$(cd "$PLUGIN_DIR_INPUT" && pwd)" || true
+if [[ ! -d "$PLUGIN_DIR" ]]; then
+    # try to find the plugin in the directory of pack-plugin
+    cd "$(dirname $0)"
+    PLUGIN_DIR="$(cd "$PLUGIN_DIR_INPUT" && pwd)" || true
+    if [[ ! -d "$PLUGIN_DIR" ]]; then
+        echo "No such plugin found: $PLUGIN_DIR_INPUT"
+        exit 2
+    fi
+fi
+PLUGIN_DIR_SRC="$(cd $PLUGIN_DIR/src && pwd)" || true
+PLUGIN_BASE="$(basename "$PLUGIN_DIR")"
+PLUGIN_ZIP_FILE="${PLUGIN_DIR}/${PLUGIN_BASE}.zip"
```

Comment:
The zip filename should not be the directory name but the name from manifest.json. I can change the validating script to enforce the directory name be the same if you want.

Reply:
AFAIK directory name here should be the same as name from manifest.json and the zip.